hs_core/task_utils.py

```diff
@@ -28,15 +39,224 @@ def _retrieve_task_id(res_id, job_dict):
+                                                                        'status': status,
+                                                                        'username': username
+                                                                        })
+        if not created:
+            if username:
+                obj.username = username
+            if name:
+                obj.name = name
+            if payload:
+                obj.payload = payload
+            obj.status = status
+            obj.save()
+
+
+def get_resource_bag_task(res_id):
+    job_name = 'hs_core.tasks.create_bag_by_irods'
+    return _retrieve_job_id(job_name, res_id)
+
+
+def get_resource_delete_task(res_id):
+    job_name = 'hs_core.tasks.delete_resource_task'
+    return _retrieve_job_id(job_name, res_id)
+
+
+def get_all_tasks(username):
+    """
+    get all tasks by a user identified by username input parameter
+    :param username: the user to retrieve all tasks for
+    :return: list of tasks where each task is a dict with id, name, and status keys
+    """
+    i = inspect()
+    act_task_list, act_task_ids = _retrieve_user_tasks(username, i.active(), 'active')
+    res_task_list, res_task_ids = _retrieve_user_tasks(username, i.reserved(), 'reserved')
+    sched_task_list, sched_task_ids = _retrieve_user_tasks(username, i.scheduled(), 'scheduled')
+    task_list = act_task_list + res_task_list + sched_task_list
+    task_ids = act_task_ids.union(res_task_ids).union(sched_task_ids)
+    task_notif_list = []
+    for obj in TaskNotification.objects.all():
+        task_notif_list.append({
+            'id': obj.task_id,
+            'name': obj.name,
+            'status': dict(TaskNotification.TASK_STATUS_CHOICES)[obj.status],
+            'payload': obj.payload
+        })
+    task_list.extend(item for item in task_notif_list if item['id'] not in task_ids)
+    return task_list
+
+
+def get_task_by_id(task_id, name='', payload='', request=None):
+    """
+    get task dict by celery task id
+    :param task_id: task id
+    :param name: task name with default being empty
+    :param payload: task payload to use. If empty, task return result will be used as payload
+    :param request: the request object
+    :return: task dict with keys id, name, status
+    """
+    result = AsyncResult(task_id)
+    status = dict(TaskNotification.TASK_STATUS_CHOICES)['progress']
+    username = request.user.username if request else ''
+    if result.ready():
+        try:
+            ret_value = result.get()
+            status = dict(TaskNotification.TASK_STATUS_CHOICES)['completed']
+            if not payload:
+                payload = ret_value
+            if name == "resource delete" and request:
+                res = get_resource_by_shortkey(ret_value)
+                res_title = res.metadata.title
+                res_type = res.resource_type
+                post_delete_resource.send(sender=type(res), request=request, user=request.user,
```

Comment:
I think we need to figure out how to call this from the task or find a different solution.  I don't like that this post delete signal relies on this method being called repeatedly in order to check the task and then send the signal.

Reply:
What if we tied into this signal
https://docs.celeryproject.org/en/latest/userguide/signals.html#task-success

And from there the post_delete_resource signal is sent.

Reply:
hmm, I looked over your comments and the reason you can't call this in the task is serialization... I'm going to rethink this a bit.  We should either be able to serialize the request or create another method for hs_tracking to take the fields of a request it needs instead of the whole request object.  This workaround distorts the readability of our codebase.

Reply:
@alvacouch - Could you chime in here?  We would like to be able to send the post_delete_resource signal from a celery task but currently are unable to because the receiver (capture_resource_delete) requires a request objects to be provided as a kwarg.  A request object is not serializable and because of this, we can't pass a request object between celery tasks and django.

Reply:
@sblack-usu @alvacouch Yes, I actually initially implemented this to send this signal from the task itself naturally for better code encapsulation and readability, but it did not work due to the serialization problem for the request object. So I ended up with the current implementation. Agreed it may be worth investigating further to see if there is a way for passing over an object to a celery task without the serialization issue.

Reply:
@sblack-usu @alvacouch Found this: https://coderbook.com/@marcus/how-to-pass-complex-data-types-to-celery-tasks/. We are currently using the default JSON serializer which does not support serializing a complicated Request data object. I believe only pickle serializer supports it, but pickle has security risk and we disabled pickle serializer a while ago.

Reply:
Interesting.  It'd be nice if we could serialize the request object as is instead of making a bunch of accommodations.  I'll try to research this before the end of the week.